parser/scanner.go

```diff
@@ -53,6 +61,21 @@ func (r Scanner) Offset() int {
 	return r.offset
 }
 
+type Position struct {
+	Line int
+	Col  int
+	File string
+}
+
+func (r Scanner) Position() Position {
+	prefix := r.src[:r.offset]
+	return Position{
+		File: r.filename,
+		Line: strings.Count(prefix, "\n") + 1,
```

Comment:
OK for now, but a later optimisation could track a slice of newline offsets to support O(log _n_) binary chop.